src/lib/components/FornaContainer.react.js

```diff
@@ -0,0 +1,258 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {FornaContainer as PreFornaContainer} from 'fornac';
+import * as R from 'ramda';
+
+/**
+ * FornaContainer is a force-directed graph that is used to visualize
+ * the secondary structure of biomolecules. It is based on the fornac
+ * library (https://github.com/ViennaRNA/fornac).
+ */
+export default class FornaContainer extends Component {
+    constructor(props) {
+        super(props);
+        this.renderNewSequences = this.renderNewSequences.bind(this);
+        this.containerRef = React.createRef();
+    }
+
+    componentDidMount() {
+        const {
+            height,
+            width,
+            nodeFillColor,
+            colorScheme,
+            customColors,
+            allowPanningAndZooming,
+        } = this.props;
+
+        this._fornaContainer = new PreFornaContainer(
+            this.containerRef.current,
+            {
+                initialSize: [width, height],
+                allowPanningAndZooming: allowPanningAndZooming,
+            }
+        );
+        // initialize the correct colors
+        this._fornaContainer.addCustomColors(customColors);
+        this._fornaContainer.changeColorScheme(colorScheme);
+
+        this.renderNewSequences();
+
+        if (nodeFillColor !== undefined) {
+            this._fornaContainer.setOutlineColor(nodeFillColor);
+        }
+    }
+
+    componentDidUpdate() {
+        this.renderNewSequences();
+    }
+
+    renderNewSequences() {
+        const {sequences} = this.props;
+
+        if (this._fornaContainer) {
+            this._fornaContainer.clearNodes();
+
+            sequences.forEach(seq => {
+                const unpackedOptions = Object.assign({}, seq.options, {
+                    sequence: seq.sequence,
+                    structure: seq.structure,
+                });
+                this._fornaContainer.addRNA(seq.structure, unpackedOptions);
+            });
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const {sequences, colorScheme} = this.props;
+
+        if (!R.equals(sequences, nextProps.sequences)) {
+            return true;
+        }
+        this._fornaContainer.addCustomColors(nextProps.customColors);
+        this._fornaContainer.changeColorScheme(colorScheme);
+
+        if (nextProps.nodeFillColor !== undefined) {
+            this._fornaContainer.setOutlineColor(nextProps.nodeFillColor);
```

Comment:
what happens if `props.nodeFillColor` is used, but `nextProps.nodeFillColor` is `undefined`?

Reply:
Since the `changeColorScheme` function (https://github.com/ViennaRNA/fornac/blob/9771d6d79d400e334bb306387d5fc795097ff94d/app/scripts/fornac.js#L690) is called before `setOutlineColor`, the component will default to the coloring as determined by the color scheme. The `nodeFillColor` prop is meant to override the `colorScheme` prop. 

So if `nextProps.nodeFillColor` is `undefined` but `props.nodeFillColor` is not, the component will end up being colored according to whatever is specified in `colorScheme`.